src/index.ts

```diff
@@ -293,7 +301,9 @@ export default function solidPlugin(options: Partial<Options> = {}): Plugin {
       // see https://github.com/vitejs/vite/discussions/5109
       const ssr: boolean = transformOptions === true || transformOptions?.ssr;
 
-      if (!/\.[jt]sx/.test(id)) return null;
+      if (!(/\.[jt]sx/.test(id) || options.extensions?.includes(getExtension(id).split('?')[0])))
+        return null;
+
```

Comment:
@high1 I'd prefer to break that into two different steps to improve readability.
Something like that:

```suggestion
      if (options.extensions) {
        const extension = getExtension(id);
        const [extensionWithoutMdFlag] = extension.split('?');
        
        if (options.extensions.includes(extensionWithoutMdFlag)) return null
      }

      if (!(/\.[jt]sx/.test(id)) return null;

```

Reply:
Sure, I have this passion for one liners, but those can be hard to read...

Reply:
Totally get it, I too have this thing, it's really a JS thing haha
The issue here is that I want anyone to be able to come in and be able to contribute without having to worry about code structure.
If we had another check to do within that if, that person would have to add `{ }`